### PR TITLE
test(e2e): added gov proposal test case to interchain accounts

### PR DESCRIPTION
**Pull request by srdtrk**
## Description

closes: #XXXX

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

## Summary by CodeRabbit

- **New Features**
	- Enhanced interchain account functionality to allow for the submission of governance proposals.

### CODE REVIEW

File: e2e/tests/interchain_accounts/base_test.go [hunk 1]
```diff
@@ -151,6 +153,53 @@ func (s *InterchainAccountsTestSuite) testMsgSendTxSuccessfulTransfer(order chan
...
+			s.Require().NotNil(govModuleAddress)
+
+			testProposal := &controllertypes.MsgUpdateParams{
+				Signer: govModuleAddress.String(),
+				Params: controllertypes.NewParams(false),
+			}
+
+			msg, err := govv1.NewMsgSubmitProposal(
+				[]sdk.Msg{testProposal},
+				sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdkmath.NewInt(100000))),
```
Comment (damiannolan) [inline e2e/tests/interchain_accounts/base_test.go:169]: We need chainB denom here afaik.

```
failed to execute interchain account transaction: deposited 100000stake, but gov accepts only the following denom(s): [atomb]: invalid deposit denom sequence 2 [36mmodule=[0mx/ibc-interchainaccounts

```

**Comment by damiannolan** [comment]
Nice, its all passing ❤️

**Comment by crodriguezvega** [comment]
@srdtrk Would you still like to merge this test?

I would probably move the steps you added to a new block (i.e. so that it's not part of the `"interchain account executes a bank transfer on behalf of the corresponding owner account"` block, since the code you added is doing something else). And potentially maybe we could just add a new test altogether for this. If you prefer, I can do these changes. Please let me know.

**Comment by dugesi** [comment]
55

**Review by damiannolan — APPROVED** [review]
💯 

Sorry about this sitting for so long!

**Pull request merged**